app/routers/register.py

```diff
@@ -0,0 +1,81 @@
+from app.internal import user
+from app.database.schemas import UserCreate
+from app.dependencies import get_db, templates
+from fastapi import APIRouter, Depends, Request
+from pydantic import ValidationError
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
+
+
+router = APIRouter(
+    prefix="",
+    tags=["register"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+async def render_registration_error(
+        db: Session, new_user: UserCreate) -> dict:
+    """After registering new user fails, defines relevant errors"""
+    db.rollback()
```

Comment:
Why should we rollback the DB first?

Reply:
There was an issue when IntegrityError is raised by sqlalchemy. 
![image](https://user-images.githubusercontent.com/71118404/107637867-7680e980-6c77-11eb-81c3-203ae66a8026.png)
rolling back is the solution i found.